wrappers/android/tools/camera/src/main/java/com/intel/realsense/camera/PreviewActivity.java

```diff
@@ -320,4 +321,38 @@ private void setOption(Sensor s, Option option, int val) {
         else
             Toast.makeText(this, "This control is not supported by this device", Toast.LENGTH_LONG).show();
     }
+
+    void resumeBackgroundTasks() {
+        resumeFwLogger();
+    }
+
+    void pauseBackgroundTasks() {
+        pauseFwLogger();
+    }
+
+    void resumeFwLogger() {
+        SharedPreferences sharedPref = getSharedPreferences(getString(R.string.app_settings), Context.MODE_PRIVATE);
+        boolean fw_logging_enabled = sharedPref.getBoolean(getString(R.string.fw_logging), false);
+        String fw_logging_file_path = sharedPref.getString(getString(R.string.fw_logging_file_path), "");
+
+
+        if (fw_logging_enabled) {
```

Comment:
What happen if `resumeFwLogger` is called twice, what will happen to the original thread?

Reply:
Well, this is not part of the API, this is part of the application.
It is not called twice, I don't think I have to check this issue.